storage/innobase/row/row0merge.cc

```diff
@@ -4338,6 +4338,12 @@ row_merge_create_index(
 		this index, to ensure read consistency. */
 		ut_ad(index->trx_id == trx->id);
 	} else {
+		/* In case we were unable to assign an undo record for this index
+		we won't free index memory object */
+		if(err == DB_TOO_MANY_CONCURRENT_TRXS)
```

Comment:
where is it freed for other err codes?

Alos leave space after if

Reply:
It happens on different places depending on the error. 
https://github.com/percona/percona-server/blob/5.7/storage/innobase/dict/dict0dict.cc#L2699 
https://github.com/percona/percona-server/blob/5.7/storage/innobase/dict/dict0dict.cc#L2728

For `DB_TOO_MANY_CONCURRENT_TRXS` it's missing. 

> Alos leave space after if
You mean a white line after the comment?

Reply:
Ideally there should be one place to free this object for all error paths. But I guess it is more complicated. The responsibility is left to callers.

So ok with your patch. Please check the style suggestion (space issue)